### PR TITLE
Add dynamic config for encoding type

### PR DESCRIPTION
**Pull request by longquanzheng**
This is to allow each domain to customize event encoding type.

PS: still fixing unit tests

### CODE REVIEW

File: common/service/dynamicconfig/constants.go [hunk 1]
```diff
@@ -308,7 +309,8 @@ const (
 	ShardUpdateMinInterval
 	// ShardSyncMinInterval is the minimal time interval which the shard info should be sync to remote
 	ShardSyncMinInterval
-
+	// EventEncodingPropertyFilteredByDomainKey is the encoding type for history events
+	EventEncodingPropertyFilteredByDomainKey
```
Comment (samarabbas) [inline common/service/dynamicconfig/constants.go]: Let's stick to the convention used by other properties and call it DefaultEventEncoding or something.

Comment (longquanzheng) [reply common/service/dynamicconfig/constants.go]: I will talk to Bowei to make sure this sticks to our convention

Comment (vancexu) [reply common/service/dynamicconfig/constants.go]: We don't need surfix `PropertyFilteredByDomainKey` in config name. Agree with samar just name it `EventEncoding` or `DefaultEncoding` etc.

Comment (longquanzheng) [reply common/service/dynamicconfig/constants.go]: I see. Suffix is only a convention in config function.

**Pull request merged**